implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -211,6 +222,26 @@ public String getRawValue(String name) {
         return getOptionalValue(name, Converters.newCollectionConverter(converter, collectionFactory));
     }
 
+    @Experimental("TODO")
+    public <T> T getConfigMapping(Class<T> klass) {
+        return getConfigMapping(klass, "");
+    }
+
+    @Experimental("TODO")
+    public <T> T getConfigMapping(Class<T> klass, String prefix) {
+        try {
+            final ConfigMappingProvider.Result result = configMappingProvider.mapConfiguration(this);
```

Comment:
We shouldn't be re-mapping the configuration on every request; it should only be done once when the configuration is built.  If validation fails, the configuration build should fail.

Reply:
Hum, and how about dynamic sources? Your mapping may be valid / invalid on built time vs lookup time depending on the sources state.

Reply:
This is why I made the mapper separate to begin with.  It is only valid at a point in time.  It is fundamentally incompatible with dynamic sources, and must always be, because an important part of the validation checking involves determining if there are required properties that are missing.  Allowing dynamicity in mapped configurations makes this impossible, because a property that is missing may appear at any time in a dynamic configuration environment.  Dynamic configuration mapping also makes map-backed mappings highly impractical, because every time any property is read, the entire configuration would have to be analyzed to determine what is present or missing at that point in time.